### DISCUSSION TITLE
Xiaomi Mijia / LUMI Aqara – sensore di apertura varchi per monitoraggio posizione tenda esterno

### DISCUSSION THREAD

**Discussion by Buff81** [indomus/forum#1607] [Hardware & Firmware]
Ciao,
vorrei provare ad utilizzare un sensore di apertura porte (possibilmente zigbee) per monitorare lo stato di una tenda da esterni elettrificata somfy: in pratica ci sono i bracci a compasso che si allargano portandosi da piegati ad allungati. Su questi vorrei montare il sensore, in modo tale da sapere quando la tenda è aperta e quando no (l'anemometro a volte me la chiude e quando torno a casa la sera trovo la sorpresa).
Quello che non trovo da nessuna parte è questo: a che distanza devono essere i due elementi del sensore per "sentirsi" ? Se rimangono non esattamente paralleli possono funzionare lo stesso? devo capire se mi tocca accroccare anche uno spessore a forma di cuneo per farli diventare paralleli.

(allego un paio di foto per chiarezza)

![IMG20220330190305](https://user-images.githubusercontent.com/89457401/161025973-f8094109-7964-4e90-b1d9-7e230e806a1f.jpg)
![IMG20220330190523](https://user-images.githubusercontent.com/89457401/161025982-41c79ae4-9086-416a-bfdf-f88685a71651.jpg)

Scopo finale

Ho già installato un trasmettitore Somfy 1810135 accoppiato a due shelly 1 in modo tale da avere i tasti fisici per azionare la tenda (prima c'era solo il telecomando, l'ho trovata così in casa acquistata) ed un controllo smart: funziona.

Vorrei quindi installare un sensore pioggia costruito come da guida di questo sito modificando un Aqara flood (https://indomus.it/progetti/realizzare-un-sensore-di-pioggia-domotico-modificandone-uno-di-allagamento/ ) e quindi farla chiudere in caso di pioggia (oggi mi è arrivata la PCB)... Successivamente vorrei pensare anche qualche logica tipo "se alle 12:00 - 13:00 - 14:00 - 15:00 - 16:00 - 17:00 - 18:00 è chiusa, e non piove, apri la tenda" :se per sbaglio tira vento, tanto si richiude da sola. 
Avere il controllo della posizione mi potrebbe anche permettere di verificare (in caso di vari tentativi di apertura) quante volte si è chiusa e facendo quindi desistere dall'insistere... Insomma, devo pensarci. Però prima di tutto devo guadagnare la conoscenza della posizione della tenda, senza non vado da nessuna parte. 
Non ho ancora installato HA, ma ho un Rasp P3b da tirare fuori dal cassetto ed una SD nuova ed il dongle ITEAD zigbee arrivato ieri, appena trovo la forza/tempo parto a divertirmi, possibilmente prima del caldo estivo.

Grazie per qualunque dritta

**Comment by Buff81** [comment]
Nessuno ha esperienza di accroccamenti simili o proposte alternative?

**Comment by Bonerserk** [comment]
Io utilizzo i sensori varchi Aqara. Sono composti dal sensore e dal pezzo più piccolo che è semplicemente un magnete. Essendo un magnete non c'è bisogno che siano paralleli. Li ho montati su porte e finestre mantenedo credo 1 - 1,5 cm tra i due pezzi. In pratica al limite, così anche un piccolo movimento attiva il sensore.
Puoi trovare il punto limite anche semplicemente ascoltando. quando il magnete si allontana senti un click nel sensore.
Oppure quando attiverai HA vedi quando è aperto o chiuso.

**Reply by Buff81** [reply]
grazie dell'info. Intanto mi sono arrivati le PCB, l'Aqara flood ed il sensore varchi: effettivamente è proprio piccolo, me l'aspettavo un po' più grande! Comunque c'ho ragionato, non si vede dalle foto, ma credo che il magnete mi convenga attaccarlo al lato della tenda sulla sbarra in alluminio in fondo (un po' di colla a caldo?) ed il sensore invece fisso sul muro (magari lo attacco ad una piastrina che fermo con un tassello): in teoria lo spazio c'è. Messo giusto la tenda chiusa ci si fermerebbe davanti.
Così tra l'altro non devo rischiare prestazioni funamboliche e pericolose...
Questo fine settimana è un casino per problemi in famiglia (da domani devo correre e fare da taxista), ma la prossima settimana mi ci metto di buzzo buono: termino l'installazione di HA su Raspberry e comincio a trafficarci. Sicuro intaserò forum e canale telegram di domande, spero di non affondare, ma non demordo facilmente.
